lib/bugsnag.ex

```diff
@@ -73,14 +74,22 @@ defmodule Bugsnag do
   end
 
   defp send_notification(body) do
-    HTTPoison.post(@notify_url, body, @request_headers)
+    release_stage = Application.get_env(:bugsnag, :release_stage)
+    notify_release_stages = Application.get_env(:bugsnag, :notify_release_stages)
+
+    if (release_stage && length(notify_release_stages) && Enum.member?(notify_release_stages, release_stage)) do
```

Comment:
It might make sense to pull this out into it's own function, and to avoid the boolean logic there (since we can pattern match on the function head).